src/PollingBlockTracker.ts

```diff
@@ -303,25 +281,59 @@ export class PollingBlockTracker
     }
     return res.result;
   }
-}
 
-/**
- * Waits for the specified amount of time.
- *
- * @param duration - The amount of time in milliseconds.
- * @param unref - Assuming this function is run in a Node context, governs
- * whether Node should wait before the `setTimeout` has completed before ending
- * the process (true for no, false for yes). Defaults to false.
- * @returns A promise that can be used to wait.
- */
-async function timeout(duration: number, unref: boolean) {
-  return new Promise((resolve) => {
-    const timeoutRef = setTimeout(resolve, duration);
-    // don't keep process open
-    if (timeoutRef.unref && unref) {
+  /**
+   * The core polling function that runs after each interval.
+   * Updates the latest block and then queues the next update.
+   */
+  private async _updateAndQueue() {
+    let interval = this._pollingInterval;
+
+    try {
+      await this._updateLatestBlock();
+    } catch (err: any) {
+      const newErr = new Error(
+        `PollingBlockTracker - encountered an error while attempting to update latest block:\n${
+          err.stack ?? err
+        }`,
+      );
+
+      try {
+        this.emit('error', newErr);
+      } catch (emitErr) {
+        console.error(newErr);
+      }
+
+      interval = this._retryTimeout;
+    }
+
+    if (!this._isRunning) {
```

Comment:
We check after in case the network is slow and the polling was stopped while the block query itself was still happening, meaning we don't want to start another timeout and continue since there would be no way to stop it since all the listeners have already gone.